plugins/jobs/src/js/types/Job.ts

```diff
@@ -87,138 +86,80 @@ type Job {
 }
 `;
 
-export function JobTypeResolver(response: MetronomeGenericJobResponse): Job {
+export function JobTypeResolver(job: MetronomeGenericJobResponse): Job {
   return {
-    id: JobFieldResolvers.id(response),
-    description: JobFieldResolvers.description(response),
-    command: JobFieldResolvers.command(response),
-    disk: JobFieldResolvers.disk(response),
-    mem: JobFieldResolvers.mem(response),
-    cpus: JobFieldResolvers.cpus(response),
-    docker: JobFieldResolvers.docker(response),
-    json: JobFieldResolvers.json(response),
-    labels: JobFieldResolvers.labels(response),
-    lastRunStatus: JobFieldResolvers.lastRunStatus(response),
-    name: JobFieldResolvers.name(response),
-    path: JobFieldResolvers.path(response),
-    jobRuns: JobFieldResolvers.jobRuns(response),
-    lastRunsSummary: JobFieldResolvers.lastRunsSummary(response),
-    scheduleStatus: JobFieldResolvers.scheduleStatus(response),
-    activeRuns: JobFieldResolvers.activeRuns(response),
-    schedules: JobFieldResolvers.schedules(response)
-  };
-}
-
-export const JobFieldResolvers = {
-  id(job: MetronomeGenericJobResponse): string {
-    return job.id;
-  },
-  description(job: MetronomeGenericJobResponse): string | null {
-    return isMetronomeJobDetailResponse(job) ? job.description : null;
-  },
-  command(job: MetronomeGenericJobResponse): string {
-    return job.run.cmd;
-  },
-  disk(job: MetronomeGenericJobResponse): number {
-    return job.run.disk;
-  },
-  mem(job: MetronomeGenericJobResponse): number {
-    return job.run.mem;
-  },
-  cpus(job: MetronomeGenericJobResponse): number {
-    return job.run.cpus;
-  },
-  activeRuns(job: MetronomeGenericJobResponse): JobRunConnection | null {
-    return isMetronomeJobDetailResponse(job)
+    activeRuns: isMetronomeJobDetailResponse(job)
```

Comment:
We could have used real field resolvers now, we would write in the job Model (like here: https://github.com/dcos/dcos-ui/blob/82b6efa19fefbfcc8a91201ad9de9ab47109023f/plugins/jobs/src/js/data/JobModel.ts#L154)

```ts
return {
  Job: {
    activeRuns(job) {
      return isMetronomeJobDetailResponse(job) ? job.activeRuns : null
    }
  }
}
```

But I think this is out of scope for this PR. Maybe we can use a spike day to make this pretty

Also Kudos for creating separate commits, this makes the review way easier

Reply:
@DanielMSchmidt can i just change that stuff to the syntax you're outlining and everything will work automagically or do i need to think throughout the process.

Also thanks for requesting a system test! I'll add one.

Reply:
If you move to this syntax it should work automatically. parent is the object returned by the parent field resolver, you also have access to args and context.